lib/java/persistance/src/main/java/org/enso/persist/PerGenerator.java

```diff
@@ -9,28 +9,35 @@
 import java.util.IdentityHashMap;
 import java.util.Map;
 import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 final class PerGenerator {
+  private static final Logger LOG = Logger.getLogger(Persistance.class.getPackageName());
 
-  static final byte[] HEADER = new byte[] {0x0a, 0x0d, 0x02, 0x0f};
+  static final byte[] HEADER = new byte[] {0x0a, 0x0d, 0x03, 0x0f};
   private final OutputStream main;
   private final Map<Object, Integer> knownObjects = new IdentityHashMap<>();
-  private final Map<Class, int[]> knownTypes = new HashMap<>();
+  private final Histogram histogram;
   private final PerMap map;
   final Function<Object, Object> writeReplace;
   private int position;
 
-  private PerGenerator(OutputStream out, int position, Function<Object, Object> writeReplace) {
+  private PerGenerator(
+      OutputStream out, Histogram histogram, int position, Function<Object, Object> writeReplace) {
     this.map = PerMap.create();
     this.main = out;
     this.writeReplace = writeReplace == null ? Function.identity() : writeReplace;
     this.position = position;
+    this.histogram = histogram;
   }
 
   static byte[] writeObject(Object obj, Function<Object, Object> writeReplace) throws IOException {
+    var histogram = LOG.isLoggable(Level.FINE) ? new Histogram() : null;
```

Comment:
The [015f4bb](https://github.com/enso-org/enso/pull/8635/commits/015f4bb09b462c275ce895aa418370500fbc3b84) change introduces typical logging code [I am used to write](https://bits.netbeans.org/dev/javadoc/org-openide-util/org/openide/util/doc-files/logging.html). The logging is disabled by default. However I have no idea how to configure it to actually log something. Trying:
```
enso$ JAVA_OPTS=-Dorg.enso.persist.Logger.level=trace
```
doesn't work, @hubertp!